dali/pipeline/operators/reader/nvdecoder/dynlink_nvcuvid.cc

```diff
@@ -206,6 +159,14 @@ CUresult CUDAAPI cuvidInit(unsigned int Flags)
 }
 
 bool cuvidInitChecked(unsigned int Flags) {
+    static std::mutex m;
+    static bool initialized = false;
+
+    std::unique_lock<std::mutex> l(m);
```

Comment:
maybe std::lock_guard, since you don't need the capabilities of std::unique_lock, but no strong opinion on that anyway

Reply:
Done